criu/image.c

```diff
@@ -211,7 +211,8 @@ int prepare_inventory(InventoryEntry *he)
 	if (get_task_ids(&crt.i))
 		return -1;
 
-	he->has_root_cg_set = true;
+	if (!opts.uid)
+		he->has_root_cg_set = true;
```

Comment:
This looks very very suspicious. We set has_root_cg_set only if uid is zero, but we always call dump_task_cgroup. If here is any reasons for this, we need to write a comment here.

When I suggested to add the --unprivileged options, I mean that we will check opts.unprivileged instead of opts.uid.